More NaN usage and create_model cleanup

karlb opened the pull request:
This PR is a grab bag of smaller cleanups, so I recommend to review commit by commit.

karlb committed:
Use NaNs in ex and ey

This allows us to skip passing edx and edy around, since we get the same
information from the NaNs.

karlb committed:
Stop using idx, idy in directvec_alg

karlb committed:
Remove useless reshape

karlb committed:
Remove useless call to compute_fd

karlb committed:
Inline update_model into create_model

It was only called in that place and duplicated multiple operations done
in create_model. Now create_model is far too large, but we can clean up
more easily from here.

karlb committed:
Extract make_individual_theos from create_model

This can be done after the main part of create_model. We'll pull it out
into a separate function. In the future we might not even call that as
part of create_model.

Holger-Bartel commented on causing/estimate.py:
```diff
@@ -20,6 +20,7 @@ def sse_orig_alg(direct, model_dat):
         direct = np.array(direct).reshape(-1)
         mx, my = utils.directmat_alg(direct, model_dat["idx"], model_dat["idy"])
         ex_hat, _ = utils.total_effects_alg(mx, my, model_dat["edx"], model_dat["edy"])
+        ex_hat = utils.nan_to_zero(ex_hat)
```
You changed ex_hat foor good not just temporarily. Was that really intended?

karlb replied:
The modified ex_hat does not leave the function, so that is fine.

Holger-Bartel commented on output/example/graphs.json:
```diff
@@ -54,23 +54,23 @@
         ],
         [
-            5.86522,
-            0.802755
```
those should not be replaced

karlb replied:
Well spotted! I missed them among all the good changes. This is fixed now along with the other unwanted NaNs.

Holger-Bartel commented on output/example/graphs.json:
```diff
@@ -54,23 +54,23 @@
         ],
         [
-            0.221734,
-            0.199683
```
those should not be replaced

Holger-Bartel commented on output/example/graphs.json:
```diff
@@ -129,8 +129,8 @@
     ],
     "exj_hat_std": [
-        0.221734,
-        0.199683
```
those should not be replaced

Holger-Bartel commented on output/example/graphs.json:
```diff
@@ -566,27 +566,27 @@
         ],
         [
-            6.1991,
-            0.746567,
```
those should not be replaced

Holger-Bartel commented on output/example/graphs.json:
```diff
@@ -652,8 +652,8 @@
     ],
     "eyj_hat_std": [
-        6.1991,
-        0.746567,
```
those should not be replaced

Holger-Bartel commented on output/example/graphs.json:
```diff
@@ -1285,16 +1285,16 @@
         ],
         [
-            0.0,
-            0.199683
```
those should not be replaced

Holger-Bartel commented on output/example/graphs.json:
```diff
@@ -1374,18 +1374,18 @@
         ],
         [
-            5.71979,
-            0.0,
```
those should not be replaced

Holger-Bartel commented on output/example/graphs.json:
```diff
@@ -1374,18 +1374,18 @@
         ],
         [
-            0.47931,
-            0.746567,
```
those should not be replaced

Holger-Bartel commented on output/example/graphs.json:
```diff
@@ -1537,12 +1537,12 @@
         ],
         [
             0.0,
-            0.802755
```
those should not be replaced

karlb committed:
Fix bad estimation results due to NaNs

This should have been part of 53215d241963826b070d11ecd72cf4630e972777
from the start.
